web/src/engine/main/src/headless/languageProcessor.ts

```diff
@@ -38,11 +38,15 @@ export class LanguageProcessor extends EventEmitter<LanguageProcessorEventMap> {
       maxRightContextCodePoints: supportsRightDeletions ? 0 : 64
     }
 
-    if(!predictiveTextWorker) {
+    if(!predictiveWorkerFactory) {
       return;
     }
 
-    this.lmEngine = new LMLayer(capabilities, predictiveTextWorker);
+    try {
+      this.lmEngine = new LMLayer(capabilities, predictiveWorkerFactory?.constructInstance());
+    } catch {
+      // We can condition on `lmEngine` being null/undefined.
+    }
```

Comment:
Ideally the try/catch would be around constructInstance() only, not around new LMLayer. Even then, if there is an error in the constructor apart from the WebWorker not being available, the catchall is going to mask the error completely. Hence, I think we should `console.warn()` on it (not sure my suggestion is perfect here).

```suggestion
    let instance: <TypeHere>;
    try {
      instance = predictiveWorkerFactory?.constructInstance();
    } catch(e) {
      // We can condition on `lmEngine` being null/undefined.
      console.warn('Web workers are not available: ' + (e ?? '').toString());
      instance = null;
    }
    if(instance) {
      this.lmEngine = new LMLayer(capabilities, instance);
    }
```